cmd/restic/backup_progress.go

```
@@ -0,0 +1,176 @@
+}
+
+// NewProgressStatus return ProgressStatus struct from scanner functions
+func NewProgressStatus(Duration time.Duration, Directories, Files, TotalBytes uint64) *ProgressStatus {
+	return &ProgressStatus{
+		ProgressType: ProgressTypeScanner,
+		Duration:     Duration,
+		Directories:  Directories,
+		Files:        Files,
+		TotalBytes:   TotalBytes,
+	}
+}
+
+// UpdateScanStatus return ProgressStatus struct from scanner functions
+func (ps *ProgressStatus) UpdateScanStatus(Duration time.Duration, Directories, Files, TotalBytes uint64) {
+	ps.ProgressType = ProgressTypeScanner
+	ps.Duration = Duration
+	ps.Directories = Directories
+	ps.Files = Files
+	ps.TotalBytes = TotalBytes
+	ps.Seconds = uint64(Duration / time.Second)
+}
+
+// UpdateProgressStatus update current progress struct with recent data
+func (ps *ProgressStatus) UpdateProgressStatus(Duration time.Duration,
+	Completed, Total, Bps, CompletedBytes uint64,
+	TotalBytes, ItemsDone, ItemsTotal, ETA, Errors uint64) {
+
+	ps.ProgressType = ProgressTypeArchive
+	ps.Duration = Duration
+
+	ps.Completed = Completed
+	ps.Total = Total
+	ps.Bps = Bps
+
+	ps.CompletedBytes = CompletedBytes
+
+	ps.TotalBytes = TotalBytes
+	ps.ItemsDone = ItemsDone
+	ps.ItemsTotal = ItemsTotal
+	ps.Seconds = uint64(Duration / time.Second)
+}
+
+// PrintJSON output of current status
+func (ps *ProgressStatus) PrintJSON() {
+	js, err := json.Marshal(ps)
+
+	if err != nil {
+		msg := fmt.Sprintf("%s\n", err)
+		Warnf(msg)
+		Exitf(100, msg)
+	}
+
+	f := bufio.NewWriter(os.Stdout)
+	fmt.Fprintf(f, "%s\n", js)
+	f.Flush()
+}
+
+// PrintScannerProgress print the scanner content
+func (ps *ProgressStatus) PrintScannerProgress() {
+	var result bytes.Buffer
+	_ = scanProgressUpdateTemplate.Execute(&result, ps)
+	PrintProgress(result.String())
+}
+
+// PrintScannerDone will print the scanner done content
+func (ps *ProgressStatus) PrintScannerDone() {
+	var result bytes.Buffer
+	_ = scanProgressDoneTemplate.Execute(&result, ps)
```

Same here, do not just ignore the error.